### ISSUE TITLE
Support includeAutomaticOptionalChainCompletions option

### ISSUE THREAD

**Issue by alxdnlnko** [neoclide/coc-tsserver#134]
Hi. There is a bug with optional chain operator completion if the `strictNullChecks` mode is enabled in the tsconfig.json.

TL;DR - for objects which are possibly null, typescript returns the the completed item with prepended `?.` symbols and vim just appends it to the previously manually added dot, and we get: `obj.?.prop`. 

Example:
```
class Some {
  a?: string

  test() {
    this.a|  <-- cursor here
  }
}
```

When I add a dot and go through the suggestions this happens:
```
this.a.|
this.a.?.length|
```

Typescript lib returns the `?.prop` completed item, and nvim wouldn't delete the dot inserted manually before. It's very annoying, and I'd prefer to insert the `?` manually, and luckily there is an option: `typescript.suggest.includeAutomaticOptionalChainCompletions`
Look here the [definition](https://github.com/microsoft/TypeScript/blob/065a996345fcfafd3c744d2a724a1ae9f31f9ab0/lib/protocol.d.ts#L2429)

But it seems that coc doesn't pass this option from the coc-config to the typescript.

Here is a coc-log:
```
2020-04-02T14:55:53.263 DEBUG (pid:22043) [events] - Event: MenuPopupChanged [
  {
    col: 18,
    row: 5,
    scrollbar: false,
    completed_item: {


      word: '?.length',


      menu: '[TSC]',
      user_data: '{"cid":1585806951,"source":"tsserver-typescript","index":16}',
      info: '',
      kind: 'm',
      abbr: 'length'
    },
    width: 15,
    height: 1,
    size: 1
  },
  4
]
```

**Comment by chemzqm** [comment]
It's a new feature, not supported yet.

**Comment by alxdnlnko** [comment]
@chemzqm hi! just updated the extension to check your fix, and you made it wrong, sorry :(

> "typescript.suggest.includeAutomaticOptionalChainCompletions": {
>           "type": "boolean",
>           "default": true,
>           "description": "Enable/disable showing completions on potentially undefined values that insert an optional chain call. Requires TS 3.7+ and strict null checks to be enabled.",

It must NOT disable the completion on such values. Completion must be triggered as usual after the dot, but **this option must be passed to typescript engine to make it stop adding the '?.' to the suggestions**.

```
this.a.|
this.a.?.length|  <-- this is what we were getting in the previous versions
this.a.length|  <-- we must get this results with the option set to 'false'
```

But with the recent update the completion is not triggered at all (for both 'true' and 'false' values), until you type `this.a?.`, but you don't see any errors while you are in the INSERT mode, so it hard to understand why the completion is not triggered, until you go back to the NORMAL mode. So it made the experience even worse, sorry. Have to disable the strict mode again, but it is very useful, so I hope you can fix this issue.

The initial intention was to disable the wrong `.?.` insertions in suggestions but not the whole completion. You must get the regular `this.a.length` completion (without auto-inserted ? sign), and after you go to the NORMAL mode, you'll see the hint that `this.a` could be null (it works already), so you can fix it yourself explicitly.

**Comment by chemzqm** [comment]
Looks like you're using old version of coc-tsserver.

**Comment by alxdnlnko** [comment]
my version is 1.5.1

**Comment by chemzqm** [comment]
Fixed on latest release branch of coc.nvim.

**Comment by chemzqm** [comment]
The completion is triggered, but filterText could be wrong.

**Comment by alxdnlnko** [comment]
Oh, thanks. Sorry for troubling you
